spec/models/miq_provision_virt_workflow_spec.rb

```diff
@@ -71,6 +71,27 @@
     end
   end
 
+  context "#update_requester_from_parameters" do
+    let(:user_new) { FactoryGirl.create(:user_with_email) }
+    let(:data_new_user) { {:user_name => user_new.name} }
+    let(:data_no_user) { {} }
+    let(:current_user) { FactoryGirl.create(:user_with_email) }
+
+    it "finds and sets a new user if one is passed in" do
+      expect(User).to receive(:lookup_by_identity).and_return(user_new).once
+      expect(MiqProvisionVirtWorkflow.update_requester_from_parameters(data_new_user, current_user)).to eq user_new
+    end
+
+    it "returns the original user if a new one is not passed in" do
+      expect(MiqProvisionVirtWorkflow.update_requester_from_parameters(data_no_user, current_user)).to eq current_user
```

Comment:
@syncrou Can we include a check that `lookup_by_identity` is never called in this case?

Since the `data_no_user` is only referenced in this one test it could be a local variable.